tutorials/navigation/navigation_optimizing_performance.rst

```diff
@@ -0,0 +1,116 @@
+
+Never scale source geometry with nodes. Not only can it result in a lot of precision errors with wrongly matched vertices and edges but also some scaling only exists as visuals and not in the actual parsed data.
+E.g. if a mesh is downscaled visually in the Editor, e.g. the scale set to 0.001 on a MeshInstance, the mesh still requires a gigantic and very complex voxel grid to be processed for the baking.
+
+Performance problems with NavigationAgent path queries
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. tip::
+
+    Avoid unnecessary path resets and queries every frame in NavigationAgent scripts.
+    
+    Avoid updating all NavigationAgent paths in the same frame.
+
+Logical errors and wasteful operations in the custom NavigationAgent scripts are very common causes of performance issues, e.g. watch out for resetting the path every single frame.
+By default NavigationAgents are optimized to only query new paths when the target position changes, the navigation map changes or they are forced too far away from the desired path distance.
+
+E.g. when AI should move to the player the target position should not be set to the player position every single frame as this queries a new path every frame.
+Instead, the distance from the current target position to the player position should be compared and only when the player has moved too far away a new target position should be set.
+
+Do not check beforehand if a target position is reachable every frame. What looks like an innocent check is the equivalent of an expensive path query behind the scene.
+If the plan is to request a new path anyway should the position be reachable a path should be queried directly.
+By looking at the last position of the returned path and if that position is in a "reachable" distance to the checked position it answers the "is this position reachable?" question.
+This avoids doing the equivalent of two full path queries every frame for the same NavigationAgent.
+
+Divide the total number of NavigationAgents into update groups or use random timers so that they do not all request new paths in the same frame.
+
+Performance problems with the actual path search
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. tip::
+
+    Optimize overdetailed navigation meshes by reducing the amount of polygons and edges.
+
+The cost of the actual path search correlates directly with the amount of navigation mesh polygons and edges and not the real size of a game world.
+If a giant game world uses very optimized navigation meshes with only few polygons that cover large areas performance will be a non-issue.
+If the opposite of optimization is done, e.g. the game world is splintered into very small navigation meshes that each have tiny polygons (TileMap *cough*) the pathfinding performance will go down the drain quickly.
```

Comment:
```suggestion
If the game world is splintered into very small navigation meshes that each have tiny polygons (TileMap *cough*) the pathfinding performance will go down the drain quickly.
```
I think thats clearer, although I have to admit I had to chuckle at "If the opposite of optimization is done" - ah yes, as we all like to do. :P